docs/stories/WebTools.md

```diff
@@ -0,0 +1,53 @@
+# Web Tools
+
+## Configure web application directory in WTP configurations for web application projects
+
+### Motivation
+The `ide-integration.md` document in Gradle core specifies a story ["Expose web application directory"](...). The goal is to define an initial end-to-end framework for Buidship/WTP integration, and to
+communicate the location of the web application directory to WTP.
+
+### Estimate
+
+- ? day(s)
+
+### Implementation
+- Create branch in buildship repository `wtp-integration`.
+- Define `org.eclipse.buildship.wtp` plugin in `wtp-integration`.
+- Define `org.eclipse.buildship.wtp.test` plugin in `wtp-integration`.
+- Wait for corresponding story in TAPI to be completed.
+- Upgrade Tooling Commons (if necessary).
+- Define configuration class in `org.eclipse.buildship.wtp`, either configurator class, or a similar class to `DefaultWorkspaceGradleOperations`.
+    - Retrieve Web application model from TAPI
+    - If a project is a web application project (model will be `null` if project doesn't apply `war` plugin)
+        - Add Java and Dynamic Web Facets to project, which will generate component file
+        - Configure web app dir location in component file
+    - Otherwise, do nothing
+- Invoke WTP configuration from Buildship core
+
+### Test cases
+- Configuration only invoked on web application projects
+- Java facet is added to project
+- Dynamic Web Facet is added to project
+- If Java Facet already on project but has incorrect version, version is updated to correct version based on source Java version
+- If Dynamic Web Facet already on project but has incorrect version, version is updated to correct version
+- Default Java Facet version is x.
+- Default Dynamic Web Facet version is x.
+- If Java Facet exists on project with correct version, Facet is not affected.
+- If Dynamic Web Facet exists on project with correct version, Facet is not affected.
+- Custom web application directory location is communicated successfully to component file, file exists.
+- Default web application directory location is communicated successfully to component file, file exists.
+- Custom web application directory location is communicated successfuly to component file, file does not exist.
+- Default web application directory location is communicated successfully to component file, file does not exist.
+
```

Comment:
> But it turns out some people want to always add extra stuff in their deployment assembly page, and keep that over mutiple config refreshes.

Is this something that we should also try to do? In that case, it would be beneficial to have a test case to make sure that a previous .component file isn't overwritten. However, I'm not certain that the web application directory entry is the best way to test this, given that the only extra attribute that you've specified is the `tag` attribute.

For instance, if we want to test that the previous version of the webappdir entry is persisted, and say Gradle specifies `src/main/ian` as the webAppDir, then we would be testing for 
```
<wb-resource deploy-path="/" source-path="/src/main/webapp" tag="defaultRootSource"/>
```
to be changed to
```
<wb-resource deploy-path="/" source-path="/src/main/ian" tag="defaultRootSource"/>
```
but this doesn't test whether the .component file is updated.

Please correct me if I am wrong.

This does bring up another test though, the default webAppDir name shouldn't be propagated to the .component file if the .component file specifies a web app dir. However, the Gradle model is the truth of the build, so should we propagate the default value then?

Is there any case where the web application directory is not specified in the component file? Would this be an error, or would we add the default webAppDir in?